master/buildbot/steps/source/git.py

```diff
@@ -251,46 +258,35 @@ def fresh(self):
         yield self._updateSubmodule()
         yield self._cleanSubmodule()
 
+    @defer.inlineCallbacks
     def copy(self):
-        cmd = remotecommand.RemoteCommand('rmdir', {'dir': self.workdir,
-                                                    'logEnviron': self.logEnviron,
-                                                    'timeout': self.timeout, })
-        cmd.useLog(self.stdio_log, False)
-        d = self.runCommand(cmd)
+        yield self.runRmdir(self.workdir, abandonOnFailure=False,
+                            timeout=self.timeout)
 
         old_workdir = self.workdir
         self.workdir = self.srcdir
-        d.addCallback(self.mode_incremental)
-
-        @d.addCallback
-        def copy(_):
-            cmd = remotecommand.RemoteCommand('cpdir',
-                                              {'fromdir': self.srcdir,
-                                               'todir': old_workdir,
-                                               'logEnviron': self.logEnviron,
-                                               'timeout': self.timeout, })
-            cmd.useLog(self.stdio_log, False)
-            d = self.runCommand(cmd)
-            return d
 
-        @d.addCallback
-        def resetWorkdir(_):
+        try:
+            yield self.mode_incremental()
+            cmd = buildstep.RemoteCommand('cpdir',
+                                          {'fromdir': self.srcdir,
+                                           'todir': old_workdir,
+                                           'logEnviron': self.logEnviron,
+                                           'timeout': self.timeout, })
+            cmd.useLog(self.stdio_log, False)
+            yield self.runCommand(cmd)
```

Comment:
wouldn't it better to say here

    yield self.runCommand(cmd, workdir= self.srcdir)